docs/docs/location-data-from-props.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Location Data from Props"
+---
+
+## What is location data
+
+Sometimes it can be helpful to know exactly what your app's browser URL is at any given stage. Using [@reach/router](https://github.com/reach/router) for [client-side routing](/docs/glossary#client-side) in Gatsby, `location` data represents where the app is currently, where you'd like it to go, and other helpful information. The `location` object is never mutated but `reach@router` makes it helpful to determine when navigation happens.
+
+```js
+{
+  key: 'ac3df4', // does not populate with a HashHistory!
+  pathname: '/somepage',
+  search: '?someurlquery=searching-string',
+  hash: '#about',
+  state: {
+    [userDefined]: true
+  }
+}
+```
+
+## HashHistory
```

Comment:
```suggestion
### HashHistory
```
This is sort-of an extra tidbit about what location data is, so it makes sense to be structured under that heading in the page hierarchy.